lib/person.rb

```diff
@@ -0,0 +1,58 @@
+require_relative 'nameable'
+
+class Person < Nameable
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    super()
+    @id = Random.rand(1..9999)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  attr_reader :id
+  attr_accessor :name, :age
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
+
+  def correct_name
+    @name
+  end
+
+  private
+
+  def of_age?
+    @age >= 18
+  end
+end
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
```

Comment:
- [ ] [OPTIONAL]
It's a best practice to mark `nameable` with `attr_accessor`, to give an indication that we intend to both _read_ and _write_ this variable. Let's endeavor to improve upon the same.